### ISSUE TITLE
[java] Incorrect variable name in violation

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#6127]
> Also notice that `${variableName}` is not being replaced by a value in the warning message. 

 _Originally posted by @cowwoc in [#5750](https://github.com/pmd/pmd/issues/5750#issuecomment-2889209156)_

Reproducer example:

Rule: [AvoidUsingHardCodedIP](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#avoidusinghardcodedip)

Code sample:
```java
import java.net.InetAddress;
class TestCase {
    InetAddress address = InetAddress.getByName("0.0.0.0");
}
```

Expected message: "Do not hard code the IP address address"

Actual message: "Do not hard code the IP address ${variableName}"
